### PR TITLE
Detail pages with comment for customers and projects

### LINKED ISSUE #1238: Better workflow when creating project and team
**Is your feature request related to a problem? Please describe.**
Only creating teams can controll access to projects, but in the workflow this are 2 different things.
Next problem: Many projects = many teams. Which team belongs to which project?

**Describe the solution you'd like**
1. When creating a new project set a checkbutton right of the name of the project for automatically creating an empty team.
Simple solution: Team name = project name
2. If checkbutton is set automatically change to team window for editing team members when project is saved.

With this solution new created projects are hidden for all users!

### LINKED ISSUE #1247: Show team assignment on project page
Creating a new project and assigning teams is always a two-step procedure.
It would be more convenient, if the team assignments were displayed and could be edited directly from the project page.

### LINKED ISSUE #1374: Report function for project progress
In larger teams, it would be helpful if important informations could be recorded.
These reports should contain the creator and date.
This could be a large text field that is assigned to the creator, but visible to all team members.
The entire report should be exportable / printable.

This function could be part of the meta fields.

### PR DESCRIPTION
**Pull request by kevinpapst**
## Description

Can be tested at https://demo-branch.kimai.org/

Fixes #1247 
Fixes #1238  
Fixes #1374 

Biggest changes: 
- new detail pages for customer and projects - click on an entry in the list views
- reduced data in list views (user, customer, project, activity) for better responsiveness & visual appearance on smaller screens and boxed layouts

The new "detail page":
- display all available fields (depending on some new permissions)
- displays the budget report
- lets you configure team access (one click to add a default team) 
- adds a "comment" section - these comments allow arbitrary text length and support markdown formatting and auto link formatting, they can be pinned to the top, can be deleted and they contain the creation date and author

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] I verified that my code applies to the guidelines (`composer kimai:code-check`)
- [ ] I updated the documentation (see [here](https://github.com/kimai/www.kimai.org/tree/master/_documentation))
- [x] I agree that this code is used in Kimai and will be published under the [MIT license](https://github.com/kevinpapst/kimai2/blob/master/LICENSE)

### CODE REVIEW

**Comment by HeinzWuert** [comment]
A project creator is usually admin and not teamleader.
Wouldn't it be better to automatically set only the teamleader instead of the creator?

**Comment by kevinpapst** [comment]
@HeinzWuert there is exactly one case where I could automatically find a (probably correct) team leader: if a customer has exactly one team assigned. But in this case a team is not required, as the customer is already assigned to that team and the existing teamlead is inherited from the customer.

Additionally: changing the teamlead is a matter of seconds. You can access all screens directly from the project detail page. Click on the table row with the team - an edit dialog opens in a modal. Change the teamlead, hit save.

But that brings me to a new idea: show the permissions from the customer on the project detail page, so admins can see whether the project is public accessible (and needs a team) or if the access is already limited.

**Comment by HeinzWuert** [comment]
Since projects with us always have to be tied to a team, every function is helpful in preventing public projects.

**Comment by kevinpapst** [comment]
Added the "unlocked" icon next to the customer, if the customer has no teams assigned. Also the wording in the "team permissions" box for projects without team changed to:
"Das Projekt ist nur eingeschränkt sichtbar, da Team Berechtigungen für den Kunden gesetzt sind"

File: templates/project/embed_activities.html.twig [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                        <td>{{ activity.name }}</td>
+                        <td class="actions">{{ actions.activity(activity, 'custom') }}</td>
+                    </tr>
+                {% endfor %}
+                </tbody>
+            </table>
+        {% endblock %}
+    {% endembed %}
+    <script type="text/javascript">
+        KimaiPaginatedBoxWidget.create('#activity_list_box');
```
Comment (kevinpapst) [inline templates/project/embed_activities.html.twig]: kimai.initialized?

File: templates/customer/embed_projects.html.twig [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                        <td>{{ project.comment|comment2html }}</td>
+                        <td class="actions">{{ actions.project(project, 'custom') }}</td>
+                    </tr>
+                {% endfor %}
+                </tbody>
+            </table>
+        {% endblock %}
+    {% endembed %}
+    <script type="text/javascript">
+        KimaiPaginatedBoxWidget.create('#project_list_box');
```
Comment (kevinpapst) [inline templates/customer/embed_projects.html.twig]: kimai.initialized?

File: src/Controller/TeamController.php [hunk 3]
```diff
@@ -93,6 +92,40 @@ public function editAction(Team $team, Request $request)
         return $this->renderEditScreen($team, $request);
     }
 
+    /**
+     * @Route(path="/{id}/edit_member", name="admin_team_member", methods={"GET", "POST"})
+     * @Security("is_granted('edit', team)")
+     */
+    public function editMemberAction(Team $team, Request $request)
```
Comment (kevinpapst) [inline src/Controller/TeamController.php]: add test

File: config/packages/kimai.yaml [hunk 4]
```diff
@@ -86,17 +86,21 @@ kimai:
...
+            CUSTOMERS_ALL_TEAM: ['view_team_customer','edit_team_customer','budget_team_customer','comments_team_customer','details_team_customer']
+            CUSTOMERS_TEAMLEAD: ['view_teamlead_customer','edit_teamlead_customer','budget_teamlead_customer','comments_teamlead_customer','details_teamlead_customer']
             INVOICE: ['view_invoice','create_invoice']
             INVOICE_TEMPLATE: ['manage_invoice_template']
             TIMESHEET: ['view_own_timesheet','start_own_timesheet','stop_own_timesheet','create_own_timesheet','edit_own_timesheet','export_own_timesheet','delete_own_timesheet']
             TIMESHEET_OTHER: ['view_other_timesheet','start_other_timesheet','stop_other_timesheet','create_other_timesheet','edit_other_timesheet','export_other_timesheet','delete_other_timesheet']
             PROFILE: ['view_own_profile','edit_own_profile','password_own_profile','preferences_own_profile','api-token_own_profile']
-            PROFILE_OTHER: ['view_other_profile','edit_other_profile','delete_other_profile','password_other_profile','roles_other_profile','preferences_other_profile','api-token_other_profile','teams_other_profile']
```
Comment (kevinpapst) [inline config/packages/kimai.yaml]: add delete to migration

File: config/packages/kimai.yaml [hunk 5]
```diff
@@ -107,10 +111,10 @@ kimai:
             SINGLE_USER: ['view_team_member','budget_team_project']
             SINGLE_TEAMLEAD: ['view_rate_own_timesheet','view_rate_other_timesheet','hourly-rate_own_profile','view_team_member']
             SINGLE_ADMIN: ['hourly-rate_own_profile','edit_exported_timesheet','teams_own_profile','view_team_member']
-            SINGLE_SUPER_ADMIN: ['hourly-rate_own_profile','hourly-rate_other_profile','delete_own_profile','roles_own_profile','system_information','system_configuration','plugins','edit_exported_timesheet','teams_own_profile','view_team_member']
```
Comment (kevinpapst) [inline config/packages/kimai.yaml]: add delete to migration

**Pull request merged**